lib/io/adc.cpp

```diff
@@ -40,9 +40,8 @@ std::optional<std::uint16_t> Adc::resetAndRead4(const int file_descriptor)
     = read(file_descriptor,
            &read_buffer,
            sizeof(read_buffer));  // actually consume new data, changes value in buffer
-  if (num_bytes_read != sizeof(read_buffer)) {
-    log_.log(core::LogLevel::kFatal, "Failed to read 4 bytes");
-
+  if (num_bytes_read < 2) {       // 2 bytes minimum as value ranges from 1 to 4 digits
```

Comment:
i feel like the use of the word digits here is confusing; what base are you talking about? decimal? binary? hex? maybe put the exact range instead?